components/selection/selection-observer-mixin.js

```diff
@@ -51,26 +56,27 @@ export const SelectionObserverMixin = superclass => class extends superclass {
 	updated(changedProperties) {
 		super.updated(changedProperties);
 
-		if (!changedProperties.has('selectionFor')) return;
-
-		if (this._selectionForObserver) this._selectionForObserver.disconnect();
-		if (this._provider) this._provider.unsubscribeObserver(this);
-
-		this._updateProvider();
+		if (changedProperties.has('selectionFor')) {
+			if (this._selectionForObserver) this._selectionForObserver.disconnect();
+			this._updateProviderFor();
 
-		this._selectionForObserver = new MutationObserver(() => {
-			this._updateProvider();
-		});
+			if (this.selectionFor) {
+				this._selectionForObserver = new MutationObserver(() => this._updateProviderFor());
 
-		this._selectionForObserver.observe(this.getRootNode(), {
-			childList: true,
-			subtree: true
-		});
+				this._selectionForObserver.observe(this.getRootNode(), {
+					childList: true,
+					subtree: true
+				});
+			}
+		}
```

Comment:
This is mostly unchanged from the original logic, with two exceptions:

- `unsubscribeObserver` is moved into `_updateProvider`
- `_updateProvider` now requires an element to be passed to it, so we call `_updateProviderFor` here which computes that element first